### PR TITLE
Features/solarcollector

### PR DESCRIPTION
**Pull request by c-moeller**
This PR is related to the issue https://github.com/oemof/oemof/issues/650

Not working yet.

### CODE REVIEW

**Comment by c-moeller** [comment]
@FranziPl please wait with your review. We have to clarify the data issue first. I need a running example to continue.

**Comment by c-moeller** [comment]
@FranziPl you can start your review. There are still some style errors even though I used black. I will check them later (have a meeting now).

File: src/oemof/thermal/flat_plate_collector.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    c_2,
+    temp_collector_inlet,
+    delta_temp_n,
+    date_col='date',
+    irradiance_global_col='ghi',
+    irradiance_diffuse_col='dhi',
+    temp_amb_col='temp_amb',
+):
+    """
+    Calculates collectors efficiency and irradiance of a flat plate collector.
```
Comment (FranziPl) [inline src/oemof/thermal/flat_plate_collector.py]: Calculates collectors heat, efficiency and irradiance of a flat plate collector.

File: src/oemof/thermal/flat_plate_collector.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    ..flat_plate_precalc_equation:
+
+    :math:`Q_{coll} = E_{coll} \cdot \eta_C`
+
+    Parameters
+    ----------
+    df: dataframe
+        Holds values for time, the global and diffuse horizontal irradiance and
+        the ambient temp (in Celsius degrees).
```
Comment (FranziPl) [inline src/oemof/thermal/flat_plate_collector.py]: the ambient temperature (in Celsius degrees).

File: docs/solar_thermal_collector.rst [hunk 3]
```diff
@@ -3,30 +3,80 @@
 ~~~~~~~~~~~~~~~~~~~~~~~
 Solar thermal collector
 ~~~~~~~~~~~~~~~~~~~~~~~
+The precalculations for the solar thermal collector calculate the heat of the solar collector based on the direct horizontal and the diffuse irradiance and information about the collector and the location. The following scheme shows the calculation procedure.
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]: "based on global and diffuse horizontal irradiance". 
Direct and diffuse would be possible, but the function used global and diffuse irradiance

File: docs/solar_thermal_collector.rst [hunk 4]
```diff
@@ -3,30 +3,80 @@
...
+.. image:: _pics/scheme.png
+    :width: 100 %
+    :alt: scheme.png
+    :align: center
 
-.. contents::
-    :depth: 2
-    :local:
-    :backlinks: top
+The processing of the irradiance data is done by the pvlib, which calculates the direct and diffuse irradiance on the collector area according to the azimuth and tilt angle of the collector.
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]: calculates the total in-plane irradiance.

File: docs/solar_thermal_collector.rst [hunk 5]
```diff
@@ -3,30 +3,80 @@
...
+
+        :math:`\T_{collector in}` :py:obj:`temp_collector_inlet`                      Collectors inlet temperature
+
+        :math:`\T_{amb}`          :py:obj:`temp_amb`                                  Ambient temperature
+
+        :math:`\eta_0`            :py:obj:`eta_0`                                     Optical efficiency of the collector
+
+        :math:`Q_{coll}`          :py:obj:`collector_heat`                            collectors heat
+
+    ========================= =================================================== ===========
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]: delta_temp_n is missing

**Review by FranziPl — CHANGES_REQUESTED** [review]

**Comment by FranziPl** [comment]
The pictures for the documentation are missing and the table is malformed, so I coulnd't build it locally.

**Comment by c-moeller** [comment]
@FranziPl the code is ready to review... Will you have a look at it? There are still some style errors I have to resolve, but I have a meeting now. I will do it afterwards.

**Comment by jnnr** [comment]
Before merging this branch to dev, merge the current dev branch to this one get all the updates. Especially the configuration files for flake8 and pylint do not exist on this branch yet.

File: docs/solar_thermal_collector.rst [hunk 6]
```diff
@@ -4,29 +4,122 @@
...
+
+    :math:`T_{amb}`           :py:obj:`temp_amb`               Ambient temperature
+
+    :math:`\eta_0`            :py:obj:`eta_0`                  Optical efficiency of the collector
+
+    :math:`Q_{coll}`          :py:obj:`collector_heat`         collectors heat
+
+    ========================= ================================ ====================================
+
+Please see the API for all Parameters, which have to be provided. The needed dataframe
```
Comment (jnnr) [inline docs/solar_thermal_collector.rst]: lower case parameters

File: examples/flat_plate_collector/flat_plate_collector_example.py [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+# Read data for flat collector and heat demand
+path = os.path.dirname(os.path.abspath(os.path.join(__file__, '..', '..')))
+dataframe = pd.read_csv(
+    path + '/examples/flat_plate_collector/data/data_flat_collector.csv',
+    sep=';',
+)
+
+# # Read demand time series from csv file
+# # --------------------------------------------------------------
+# demand_df = pd.read_csv(
```
Comment (jnnr) [inline examples/flat_plate_collector/flat_plate_collector_example.py]: If this is unused, it should be removed.

**Comment by FranziPl** [comment]
@c-moeller What needs a second review? I will look at docs again now. If the rest need another review too, please tell me, when I can have a look at the final version.

**Comment by c-moeller** [comment]
@FranziPl yes only the docs.

File: docs/solar_thermal_collector.rst [hunk 8]
```diff
@@ -4,29 +4,122 @@
...
+    :alt: solarcollector_scheme.png
+    :align: center
+
+The processing of the irradiance data is done by the pvlib, which calculates the total
+in-plane irradiance according to the azimuth and tilt angle of the collector.
+
+The efficiency of the collector is calculated with
+
+.. include:: ../src/oemof/thermal/flat_plate_collector.py
+    :start-after:  calc_eta_c_flate_plate_equation::
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]:
```suggestion
    :start-after:  calc_eta_c_flate_plate_equation:
```
(with the second ":" it doesn't work)

File: docs/solar_thermal_collector.rst [hunk 9]
```diff
@@ -4,29 +4,122 @@
...
+    :end-before: Parameters
+
+In the end, the irradiance on the collector is multiplied with the efficiency
+to get the collectors heat.
+
+.. include:: ../src/oemof/thermal/flat_plate_collector.py
+     :start-after:  flat_plate_precalc_equation:
+     :end-before: Parameters
+
+The three values :math:`Q_{coll}`, :math:`\eta_C` and :math:`E_{coll}` are returned.
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]:
```suggestion
The three values :math:`Q_{coll}`, :math:`\eta_C` and :math:`E_{coll}` are returned in a dataframe.
```

File: src/oemof/thermal/flat_plate_collector.py [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+    collector_irradiance,
+):
+    r"""
+    Calculates collectors efficiency
+
+    ..calc_eta_c_flate_plate_equation:
+
+    :math:`\eta_C = \eta_0 - c_1 \cdot \frac{\Delta T}{E_{coll}}
+    - c_2 \cdot \frac{{\Delta T}^2}{E_{coll}}`
+    with
```
Comment (FranziPl) [inline src/oemof/thermal/flat_plate_collector.py:154]: a blank line before and after "with" makes the docu more readable.
```suggestion

    with
    
```

File: docs/solar_thermal_collector.rst [hunk 11]
```diff
@@ -4,29 +4,122 @@
...
+    :math:`T_{collector in}`  :py:obj:`temp_collector_inlet`   Collectors inlet temperature
+
+    :math:`\Delta T_n`        :py:obj:`delta_temp_n`           Temperature difference between
+                                                               collector inlet and mean temperature
+
+    :math:`T_{amb}`           :py:obj:`temp_amb`               Ambient temperature
+
+    :math:`\eta_0`            :py:obj:`eta_0`                  Optical efficiency of the collector
+
+    :math:`Q_{coll}`          :py:obj:`collector_heat`         collectors heat
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]: Collectors heat
```suggestion
    :math:`Q_{coll}`          :py:obj:`collector_heat`         Collectors heat
```

File: docs/solar_thermal_collector.rst [hunk 12]
```diff
@@ -4,29 +4,122 @@
...
+
+These arguments are used in the formulas of the function:
+
+    ========================= ================================ ====================================
+    symbol                    argument                         explanation
+    ========================= ================================ ====================================
+    :math:`E_{coll}`          :py:obj:`collector_irradiance`   Irradiance on collector after all
+                                                               losses
+
+    :math:`\eta_C`            :py:obj:`eta_c`                  collectors efficiency
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]: Collectors efficiency
```suggestion
    :math:`\eta_C`            :py:obj:`eta_c`                  Collectors efficiency
```

**Review by FranziPl — CHANGES_REQUESTED** [review]
I commented a few things, the rest looks fine to me.

File: docs/solar_thermal_collector.rst [hunk 13]
```diff
@@ -4,29 +4,122 @@
...
+
+    :math:`c_1`               :py:obj:`c_1`                    Thermal loss parameter 1
+
+    :math:`c_2`               :py:obj:`c_2`                    Thermal loss parameter 2
+
+    :math:`\Delta T`          :py:obj:`delta_t`                Temperature difference
+                                                               (collector to ambience)
+
+    :math:`T_{collector in}`  :py:obj:`temp_collector_inlet`   Collectors inlet temperature
```
Comment (FranziPl) [inline docs/solar_thermal_collector.rst]:
```suggestion
    :math:`T_{Collector, in}`  :py:obj:`temp_collector_inlet`   Collectors inlet temperature
```

**Comment by c-moeller** [comment]
@FranziPl Had to cancel one of your suggestion due to Pep8 reasons. Please approve. Thank you!

**Review by FranziPl — APPROVED** [review]

File: src/oemof/thermal/flat_plate_collector.py (hunk 10 above)
Comment (FranziPl) [inline src/oemof/thermal/flat_plate_collector.py:154]:
```suggestion

    with

```

Comment (c-moeller) [reply src/oemof/thermal/flat_plate_collector.py:154]: This does not work due to Pep8 reasons :-(

Comment (FranziPl) [reply src/oemof/thermal/flat_plate_collector.py:154]: It does. I commited it before you merged.

**Pull request merged**